### DISCUSSION TITLE
Is there a way to structure an alpine js app with laravel ?

### DISCUSSION THREAD

**Discussion by heshamelmasry77** [alpinejs/alpine#3766] [4. General]
The main reason: it is unstructured.

**Comment by ekwoka** [comment]
Are you using Livewire?

Also, what do you mean by "unstructured" and "structure"?

**Reply by heshamelmasry77** [reply]
yes I use it and I came up with this structure

 ```
|-- js/
|       |-- alpine/
|           |-- components/   # Alpine.js components
|           |   |-- dropdown.js
|           |   |-- modal.js
|           |-- store/        # Alpine.js global store/state (if needed)
|               |-- index.js
|               |-- user.js
|           |-- init.js       # Where you initialize Alpine or import components
```

**Reply by SimoTod** [reply]
Worst structure ever.
I'm joking, but it's exactly what you did on your original post. 

Alpine is not really a framework, it's just a single file js library. How you split or organise your asset is your concern.

**Reply by ekwoka** [reply]
I mean, that's a pretty common way to do it
